### PR TITLE
Add aws-checksums/0.1.5 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **aws-checksums/0.1.5**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/aws-checksums/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.12)
+project(test_package LANGUAGES C)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.c)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (madebr) [inline recipes/aws-checksums/all/test_package/CMakeLists.txt]:
```suggestion
target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
set_property(TARGET ${PROJECT_NAME} PROPERTY C_STANDARD 99)
```

**Comment by uilianries** [comment]
@madebr it's forcing fPIC: https://github.com/awslabs/aws-checksums/blob/v0.1.5/CMakeLists.txt#L87

I think you could patch that too.

**Comment by uilianries** [comment]
Same for aws-checksums-tests. There is no key for disabling its build.

File: recipes/aws-checksums/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "aws-checksums-{}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["CMAKE_MODULE_PATH"] = self.build_folder.replace("\\", "/")
```
Comment (danimtb) [inline recipes/aws-checksums/all/conanfile.py:50]: why is this required?

Comment (madebr) [reply recipes/aws-checksums/all/conanfile.py:50]: I don't remember, but it looks like it can be removed.

Comment (madebr) [inline recipes/aws-checksums/all/conanfile.py:50]:
```suggestion
```

**Comment by danimtb** [comment]
for some reason, the CLA check is not working... 😖
